Scripts/Api.cs

```diff
@@ -123,25 +151,133 @@ public static IEnumerator CreateConversation(string token, Action<string> callba
+            {
+                requestParams.Add("conversationUuid", conversationUuid);
+            }
+            if (minEventId != default)
+            {
+                requestParams.Add("minEventId", minEventId);
+            }
+
+            var request = UnityWebRequest.Get($"{BaseUrl}/play/message-history" + CreateQueryString(requestParams));
+            request.method = "GET"; 
+            request.SetRequestHeader("Authorization", $"Bearer {token}");
+
+            Logger.Log("Requesting message history...");
 
+            yield return request.SendWebRequest();
+
+            if (request.result != UnityWebRequest.Result.Success)
+            {
+                Logger.LogError("Error:" + request.error);
+                yield break;
+            }
+
+            var data = Encoding.UTF8.GetString(request.downloadHandler.data);
+
+            try
+            {
+                var deserialized = JsonConvert.DeserializeObject<GetMessageHistoryResponse>(data);
+                callback?.Invoke(deserialized);
+                Logger.Log("Message history request complete");
+            }
+            catch (Exception e)
+            {
+                Logger.LogError($"Could not get message history; {e}");
+                throw;
+            }
+
+            // Need to dispose of WebRequest to prevent following error:
+            // "A Native Collection has not been disposed, resulting in a memory leak. Enable Full StackTraces to get more details."
+            request.Dispose();
+        }
+
+        /// <summary>
+        /// Returns information about the playthrough, including emotions of the characters and saved memories.
+        /// </summary>
+        /// <param name="token">Valid play-through token.</param>
+        /// <param name="callback">Called when a valid playthrough info response is generated.</param>
+        public static IEnumerator GetPlaythroughInfo(string token, Action<GetPlaythroughInfoResponse> callback)
+        {
+            var request = UnityWebRequest.Get($"{BaseUrl}/play/playthrough-info");
+            request.method = "GET";
+            request.SetRequestHeader("Authorization", $"Bearer {token}");
+
+            Logger.Log("Requesting playthrough info...");
+
+            yield return request.SendWebRequest();
+
+            if (request.result != UnityWebRequest.Result.Success)
+            {
+                Logger.LogError("Error:" + request.error);
+                yield break;
+            }
+
+            var data = Encoding.UTF8.GetString(request.downloadHandler.data);
+
+            try
+            {
+                var deserialized = JsonConvert.DeserializeObject<GetPlaythroughInfoResponse>(data);
+                callback?.Invoke(deserialized);
+                Logger.Log("Playthrough info request complete");
+            }
+            catch (Exception e)
+            {
+                Logger.LogError($"Could not generate playthrough information; {e}");
+                throw;
```

Comment:
I feel like this should re-throw the error rather than throwing nothing (is that a thing in C#?)

(same in below endpoints)